### PR TITLE
Removed support for token environment variables support for KDD.

### PR DESCRIPTION
**Pull request by tathagatachowdhury**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: lib/apiconfig/apiconfig.go [hunk 1]
```diff
@@ -66,10 +66,10 @@ type EtcdConfig struct {
 type KubeConfig struct {
 	Kubeconfig               string `json:"kubeconfig" envconfig:"KUBECONFIG" default:""`
 	K8sAPIEndpoint           string `json:"k8sAPIEndpoint" envconfig:"K8S_API_ENDPOINT" default:""`
-	K8sKeyFile               string `json:"k8sKeyFile" envconfig:"K8S_KEY_FILE" default:""`
-	K8sCertFile              string `json:"k8sCertFile" envconfig:"K8S_CERT_FILE" default:""`
-	K8sCAFile                string `json:"k8sCAFile" envconfig:"K8S_CA_FILE" default:""`
-	K8sAPIToken              string `json:"k8sAPIToken" envconfig:"K8S_API_TOKEN" default:""`
+	K8sKeyFile               string `json:"k8sKeyFile" default:""`
```
Comment (fasaxc) [inline lib/apiconfig/apiconfig.go]: I think you need to set `ignored:"true"`, like you did for the new EtcdConfig fields

Comment (tathagatachowdhury) [reply lib/apiconfig/apiconfig.go]: Acknowledged and updated.

**Comment by fasaxc** [comment]
@caseydavenport was this what you had in mind?  Just wanted to check you didn't mean to _completely_ remove those config params.

File: lib/client/client_test.go [hunk 2]
```diff
@@ -121,7 +121,6 @@ kind: notCalicoApiConfig
 		"APIV1_K8S_CERT_FILE":    "baz1",
 		"APIV1_K8S_KEY_FILE":     "foo1",
 		"APIV1_K8S_CA_FILE":      "foobar1",
-		"APIV1_K8S_API_TOKEN":    "foobarbaz1",
```
Comment (fasaxc) [inline lib/client/client_test.go]: Suggest you leave the env var _input_ in place (with comment `// ignored for security reasons`) but verify that it is now ignored.

Comment (tathagatachowdhury) [reply lib/client/client_test.go]: Acknowledged and updated code.

**Pull request merged**